conduit/Data/Conduit/List.hs

```diff
@@ -631,6 +633,25 @@ consumeC =
 {-# INLINE consumeC #-}
 STREAMING0(consume, consumeC, consumeS)
 
+-- | Group a stream into chunks of a given size. The last chunk may contain
+-- fewer than n elements.
+--
+-- Subject to fusion
+--
+-- Since ???
+chunksOf :: Monad m => Int -> Conduit a m [a]
+chunksOf n =
+    start
+  where
+    start = await >>= maybe (return ()) (loop n id)
+
+    loop count rest x =
```

Comment:
May be worthwhile to use a bang pattern on `count` (i.e., `!count`) to avoid laziness. The guard below should be enough for the strictness analyzer, but better safe than sorry.